From a1d4eb7535146e7f3990bc5359214a619181921c Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <matthias.schiffer@tq-group.com>
Date: Fri, 28 Apr 2023 10:39:37 +0200
Subject: [PATCH] iMX8M: add mkimage_imx8 wrapper to redirect output

Add a wrapper that tees the stderr output to a log file, allowing for
more robust parsing for CSF / HAB block offsets.

Add cleaning of the generated log file to the SoC clean target
to prevent using wrong data in subsequent runs.

Upstream-Status: Inappropriate
Signed-off-by: Matthias Schiffer <matthias.schiffer@tq-group.com>
---
 iMX8M/mkimage_imx8 | 5 +++++
 iMX8M/soc.mak      | 3 ++-
 2 files changed, 7 insertions(+), 1 deletion(-)
 create mode 100755 iMX8M/mkimage_imx8

diff --git a/iMX8M/mkimage_imx8 b/iMX8M/mkimage_imx8
new file mode 100755
index 0000000..4810d13
--- /dev/null
+++ b/iMX8M/mkimage_imx8
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+OUTLOG=mkimage.log
+
+exec ./mkimage_imx8.real "$@" 2>&1 | tee "${OUTLOG}" >&2
diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index f1b37a4..0fd5574 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -1,4 +1,4 @@
-MKIMG = mkimage_imx8
+MKIMG = mkimage_imx8.real
 OUTIMG = flash.bin
 
 CC ?= gcc
@@ -150,6 +150,7 @@ u-boot-atf-tee.bin: u-boot.bin bl31.bin $(TEE)
 .PHONY: clean
 clean:
 	@rm -f $(MKIMG) u-boot-atf.bin u-boot-atf-tee.bin u-boot-spl-ddr.bin u-boot.itb u-boot.its u-boot-ddr3l.itb u-boot-ddr3l.its u-boot-spl-ddr3l.bin u-boot-ddr4.itb u-boot-ddr4.its u-boot-spl-ddr4.bin u-boot-ddr4-evk.itb u-boot-ivt.itb u-boot-ddr4-evk.its $(OUTIMG)
+	@rm -f mkimage.log
 
 dtb = evk.dtb
 $(dtb):
